src/app/MessageProvider.tsx

```diff
@@ -1,34 +1,47 @@
 import { shouldPolyfill } from '@formatjs/intl-pluralrules/should-polyfill';
-import React from 'react';
-import { createIntl, createIntlCache, RawIntlProvider } from 'react-intl';
+import React, { useState } from 'react';
+import { RawIntlProvider } from 'react-intl';
+import { useSelector } from 'react-redux';
+import { book as bookSelector } from './content/selectors';
 import { useServices } from './context/Services';
-import enMessages from './messages/en';
+import { pathname as pathSelector } from './navigation/selectors';
 
-// https://formatjs.io/docs/polyfills/intl-pluralrules/#dynamic-import--capability-detection
-async function polyfill(locale: 'en') {
-  if (shouldPolyfill()) {
-    await import('@formatjs/intl-pluralrules/polyfill');
-  }
+// tslint:disable-next-line:variable-name
+const MessageProvider = (props: { children?: React.ReactNode }) => {
+  const book = useSelector(bookSelector);
+  const currentPath = useSelector(pathSelector);
+  const [polyfillLoaded, setPolyfillLoaded] = useState(false);
+
+  // https://formatjs.io/docs/polyfills/intl-pluralrules/#dynamic-import--capability-detection
+  async function polyfill(locale: string | undefined) {
+    if (!locale) {
+      return;
+    }
 
-  // boolean added by the polyfill
-  if ((Intl.PluralRules as (typeof Intl.PluralRules & {polyfilled?: boolean})).polyfilled) {
-    await import(`@formatjs/intl-pluralrules/locale-data/${locale}`);
+    if (shouldPolyfill()) {
+      await import('@formatjs/intl-pluralrules/polyfill');
+    }
+
+    // boolean added by the polyfill
+    if ((Intl.PluralRules as (typeof Intl.PluralRules & { polyfilled?: boolean })).polyfilled) {
+      await import(`@formatjs/intl-pluralrules/locale-data/${locale}`);
+      setPolyfillLoaded(true);
+    }
   }
-}
 
-polyfill('en');
+  const lang = React.useMemo(() => {
+    return currentPath === '/' ? 'en' : book?.language;
+  }, [book, currentPath]);
 
-const cache = createIntlCache();
+  const intl = useServices().intl.getIntlObject(lang);
 
-export const intl = createIntl({
-  locale: 'en',
-  messages: enMessages,
-}, cache);
+  polyfill(lang);
 
-// tslint:disable-next-line:variable-name
-const MessageProvider: React.FC = (props) =>
-  <RawIntlProvider value={useServices().intl}>
-    {props.children}
-  </RawIntlProvider>;
+  return (lang && !shouldPolyfill()) || (lang && polyfillLoaded) ? (
```

Comment:
if the polyfill function was separate you wouldn't need the extra condition here